sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -94,8 +95,12 @@ sealed trait Partitioning {
    */
   def compatibleWith(other: Partitioning): Boolean
 
-  /** Returns the expressions that are used to key the partitioning. */
-  def keyExpressions: Seq[Expression]
+  /**
+   * Returns true iff we can say that the partitioning scheme of this [[Partitioning]]
```

Comment:
This confusion has been significantly lessened by the removal of `compatibleWith` in my other patch.